src/store/reducers/todo.js

```diff
@@ -36,6 +37,16 @@ export const todoReducer = (state = initialTodoState, action) => {
       return { ...state, tasks: existingTasks };
     }
 
+    case EDIT_TASK: {
+      const existingTasks = [...state.tasks];
+      const selectedTaskId = findTaskIndexById(
+        action.payload.taskId,
+        existingTasks
+      );
```

Comment:
finding index of a task shouldn't mutate the original collection, hence copy is unnecessary.

```suggestion
      const selectedTaskId = findTaskIndexById(
        action.payload.taskId,
        state.tasks
      );
```